### PR TITLE
redesign and rewrite deployment document

### PR DESCRIPTION
**Pull request by dailidong**

### CODE REVIEW

File: docs/zh-cn/1.2.0/user_doc/hardware-environment.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+| 操作系统       | 版本         |
+| :----------------------- | :----------: |
+| Red Hat Enterprise Linux | 7.0 及以上   |
+| CentOS                   | 7.0 及以上   |
+| Oracle Enterprise Linux  | 7.0 及以上   |
+| Ubuntu LTS               | 16.04 及以上 |
+
+> **注意：**
+>以上 Linux 操作系统可运行在物理服务器以及 VMware、KVM、XEN 主流虚拟化环境上。
+
```
Comment (Jave-Chen) [inline docs/zh-cn/1.2.0/user_doc/hardware-environment.md]: Better to add dependency, such as: database and zookeeper.

File: docs/zh-cn/1.2.0/user_doc/hardware-environment.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+### 生产环境
+
+| **CPU** | **内存** | **硬盘类型** | **网络** | **实例数量** |
+| --- | --- | --- | --- | --- |
+| 4核+ | 8 GB+ | SAS | 千兆网卡 | 1+ |
+
+> **注意：**
+> - 以上建议配置为部署 DolphinScheduler 的最低配置，生产环境强烈推荐使用更高的配置。
+> - 硬盘大小配置建议 50GB+ ，系统盘和数据盘分开。
+
```
Comment (Jave-Chen) [inline docs/zh-cn/1.2.0/user_doc/hardware-environment.md]: Better to add number of node, how many minimum nodes for master and worker .

File: docs/zh-cn/1.2.0/user_doc/standalone-deployment.md [hunk 3]
```diff
@@ -0,0 +1,451 @@
...
+
+# 1、后端部署
+
+### 1.1 : 基础软件安装(必装项请自行安装)
+
+ * PostgreSQL (8.2.15+) or Mysql (5.6或者5.7系列)  :  两者任选其一即可
+ * [JDK](https://www.oracle.com/technetwork/java/javase/downloads/index.html) (1.8+) :  必装，请安装好后在/etc/profile下配置 JAVA_HOME 及 PATH 变量
+ * ZooKeeper (3.4.6+) ：必装 
+ * Hadoop (2.6+) or MinIO ：选装， 如果需要用到资源上传功能，针对单机可以选择本地文件目录作为上传文件夹(此操作不需要部署Hadoop)；当然也可以选择上传到Hadoop or MinIO集群上
+
```
Comment (Jave-Chen) [inline docs/zh-cn/1.2.0/user_doc/standalone-deployment.md]: MySQL is also supported by ds

File: docs/zh-cn/1.2.0/user_doc/standalone-deployment.md [hunk 4]
```diff
@@ -0,0 +1,451 @@
...
+# 修改目录权限，使得部署用户对dolphinscheduler-backend目录有操作权限  
+chown -R dolphinscheduler:dolphinscheduler dolphinscheduler-backend
+```
+
+```
+ 注意：
+ - 因为任务执行服务是以 sudo -u {linux-user} 切换不同linux用户的方式来实现多租户运行作业，所以部署用户需要有 sudo 权限，而且是免密的。初学习者不理解的话，完全可以暂时忽略这一点
+ - 如果发现/etc/sudoers文件中有"Default requiretty"这行，也请注释掉
+ - 如果用到资源上传的话，还需要给该部署用户分配操作`本地文件系统或者HDFS或者MinIO`的权限
+```
```
Comment (Jave-Chen) [inline docs/zh-cn/1.2.0/user_doc/standalone-deployment.md]: above sh command needs executed by root

File: docs/zh-cn/1.2.0/user_doc/standalone-deployment.md [hunk 5]
```diff
@@ -0,0 +1,451 @@
...
+
+- 进入数据库命令行窗口后，执行数据库初始化命令，设置访问账号和密码。**注: {user} 和 {password} 需要替换为具体的数据库用户名和密码** 
+
+``` mysql
+mysql> CREATE DATABASE dolphinscheduler DEFAULT CHARACTER SET utf8 DEFAULT COLLATE utf8_general_ci;
+mysql> GRANT ALL PRIVILEGES ON dolphinscheduler.* TO '{user}'@'%' IDENTIFIED BY '{password}';
+mysql> GRANT ALL PRIVILEGES ON dolphinscheduler.* TO '{user}'@'localhost' IDENTIFIED BY '{password}';
+mysql> flush privileges;
+```
+
```
Comment (Jave-Chen) [inline docs/zh-cn/1.2.0/user_doc/standalone-deployment.md]: add command for PostgreSQL, and create tables command.

File: docs/zh-cn/1.2.0/user_doc/standalone-deployment.md [hunk 6]
```diff
@@ -0,0 +1,451 @@
...
+    #export SPARK_HOME1=/opt/soft/spark1
+    export SPARK_HOME2=/opt/soft/spark2
+    export PYTHON_HOME=/opt/soft/python
+    export JAVA_HOME=/opt/soft/java
+    export HIVE_HOME=/opt/soft/hive
+    export FLINK_HOME=/opt/soft/flink
+    export PATH=$HADOOP_HOME/bin:$SPARK_HOME2/bin:$PYTHON_HOME:$JAVA_HOME/bin:$HIVE_HOME/bin:$PATH:$FLINK_HOME/bin:$PATH
+    
+    ```
+
```
Comment (Jave-Chen) [inline docs/zh-cn/1.2.0/user_doc/standalone-deployment.md]: if you can't find .dolphinscheduler_env.sh, run ls -a

File: docs/zh-cn/1.2.0/user_doc/standalone-deployment.md [hunk 7]
```diff
@@ -0,0 +1,451 @@
...
+     `注: 这一步非常重要,例如 JAVA_HOME 和 PATH 是必须要配置的，没有用到的可以忽略或者注释掉`
+
+- 将jdk软链到/usr/bin/java下(仍以 JAVA_HOME=/opt/soft/java 为例)
+
+    ```shell
+    sudo ln -s /opt/soft/java/bin/java /usr/bin/java
+    ```
+
+ - 修改一键部署脚本 `install.sh`中的各参数，特别注意以下参数的配置
+
```
Comment (Jave-Chen) [inline docs/zh-cn/1.2.0/user_doc/standalone-deployment.md]: these configures are very important for deploying ds. how about highlight them.

File: docs/zh-cn/1.2.0/user_doc/standalone-deployment.md [hunk 8]
```diff
@@ -0,0 +1,451 @@
...
+  *注意：如果yum没找到python-pip，也可以通过下面方式安装*
+
+```shell
+sudo curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py
+sudo python get-pip.py  # 如果是python3，使用sudo python3 get-pip.py 
+#然后
+sudo pip install kazoo;
+```
+
```
Comment (Jave-Chen) [inline docs/zh-cn/1.2.0/user_doc/standalone-deployment.md]: this step is optional

File: docs/zh-cn/1.2.0/user_doc/hardware-environment.md (hunk 1 above)
Comment (dailidong) [reply docs/zh-cn/1.2.0/user_doc/hardware-environment.md]: database and zookeeper requirement has writed in the deployment part

File: docs/zh-cn/1.2.0/user_doc/hardware-environment.md (hunk 2 above)
Comment (dailidong) [reply docs/zh-cn/1.2.0/user_doc/hardware-environment.md]: good idea.
I thought about this idea,  standalone need one machine, but not confirm how many machines they really needed in cluster mode, they need to make their own estimate according to their own business situation

File: docs/zh-cn/1.2.0/user_doc/standalone-deployment.md (hunk 3 above)
Comment (dailidong) [reply docs/zh-cn/1.2.0/user_doc/standalone-deployment.md]: PostgreSQL (8.2.15+) or Mysql (5.6或者5.7系列)  :  两者任选其一即可

File: docs/zh-cn/1.2.0/user_doc/standalone-deployment.md (hunk 4 above)
Comment (dailidong) [reply docs/zh-cn/1.2.0/user_doc/standalone-deployment.md]: yes, I will add this comment

File: docs/zh-cn/1.2.0/user_doc/standalone-deployment.md (hunk 5 above)
Comment (dailidong) [reply docs/zh-cn/1.2.0/user_doc/standalone-deployment.md]: good idea, but I'm not familiar with PostgreSQL，may be you can add when merged

File: docs/zh-cn/1.2.0/user_doc/standalone-deployment.md (hunk 6 above)
Comment (dailidong) [reply docs/zh-cn/1.2.0/user_doc/standalone-deployment.md]: good, I will add comment

File: docs/zh-cn/1.2.0/user_doc/standalone-deployment.md (hunk 7 above)
Comment (dailidong) [reply docs/zh-cn/1.2.0/user_doc/standalone-deployment.md]: good idea

File: docs/zh-cn/1.2.0/user_doc/standalone-deployment.md (hunk 8 above)
Comment (dailidong) [reply docs/zh-cn/1.2.0/user_doc/standalone-deployment.md]: which step is optional?

**Comment by dailidong** [comment]
> Does the doc work for 1.2.1 ？

1.2.0

File: docs/zh-cn/1.2.0/user_doc/standalone-deployment.md (hunk 8 above)
Comment (dailidong) [reply docs/zh-cn/1.2.0/user_doc/standalone-deployment.md]: 1.2.0 need this step, from 1.2.1, kazoo is removed

**Pull request merged**